messaging/src/main/java/io/lantern/messaging/Messaging.kt

```diff
@@ -611,6 +611,9 @@ class Messaging(
     ): Model.StoredAttachment {
         val md = Metadata.analyze(file, mimeType)
         val attachment = Model.Attachment.newBuilder().setMimeType(md?.mimeType ?: mimeType)
+        if (md.additionalMetadata != null) {
+            attachment.putAllMetadata(md.additionalMetadata)
```

Comment:
This is where we populate the metadata (that contains `duration`) into the actual attachment.